### DISCUSSION TITLE
Advanced movement in insert mode

### DISCUSSION THREAD

**Discussion by fcoury** [helix-editor/helix#4006] [General] [answered]
Hi there, new Helix user here and loving it so far.

I have a few keys that I use on my current editor (VS Code) and I would like to know if it's possible to emulate them on Helix, either stock or with a custom command.

Let me enumerate them:

1. Ability to move to the cursor to the next line even when you're in insert mode and in the middle of a line. I am currently doing `Esc`, `O`, `i` to go back to Normal, insert line below and go back to insert mode. Would love to see if there's an easy way. 

In VS Code, when I press Cmd+Enter on VS Code and I'm in the middle of a line, instead of breaking the line where I am it just inserts a new line below and moves to it:

https://user-images.githubusercontent.com/1371/192811649-f45cfad2-df2d-4d26-a9d7-ada507532e6b.mp4

2. A way to move after a closing parens, bracket, quote, etc. Auto closing them is awesome, but you need to navigate to after them manually. In VS Code if I have a line like `i_am_here( | )` where `|` is my cursor and the closing parens was added by the auto complete feature, I can press Tab and go to right after the parens, and continue writing my line - `i_am_here()|` after Tab.

https://user-images.githubusercontent.com/1371/192813075-bb2c0c7e-e71b-48a2-8655-5312800a3f97.mp4

Is this or something similar possible to achieve using Helix?

Thank you!

**Comment by Omnikar** [comment, marked as answer]
With auto pairs, you can type the closing pair and the cursor will automatically move after it.

As for the other point, doing actions like that is kind of the whole point of normal mode :P But also, you shouldn't need to hit `i` again; `o` should just open the new line and enter insert mode automatically.

**Reply by fcoury** [reply]
Thank you!

> With auto pairs, you can type the closing pair and the cursor will automatically move after it.

Yes that's what I am trying to teach my muscle memory to do 😊

> As for the other point, doing actions like that is kind of the whole point of normal mode :P But also, you shouldn't need to hit `i` again; `o` should just open the new line and enter insert mode automatically.

I wonder if you can record a macro or something similar to perform it, and assign it to Cmd+Enter on normal mode.

**Reply by Omnikar** [reply]
If you really want to have that sort of binding in insert mode, you can configure it in your `config.toml` (accessible via `:config-open`):
```toml
[keys.insert]
"A-ret" = "open_below"
```
I have it set to alt-enter here because we can't really access the cmd key from a terminal application, and ctrl-enter doesn't work due to terminal limitations.

**Reply by fcoury** [reply]
> If you really want to have that sort of binding in insert mode, you can configure it in your `config.toml` (accessible via `:config-open`):
> 
> ```toml
> 
> [keys.insert]
> 
> "A-ret" = "open_below"
> 
> ```
> 
> I have it set to alt-enter here because we cant really access the cmd key from a terminal application, and ctrl-enter doesn't work due to terminal limitations.

That is a great, will give it a shot. Thank you once again!
